code-challenges.js

```diff
@@ -6,70 +6,255 @@
+        // if the number is above the value, print this ^
+    } else if(num === boil) {
+        return `${num} is at boiling point.`
+        // if the number is =to the value, print this ^
+    } else {
+        return "You are out of range"
+        // if you type in something other than a number, print this ^ (another bug check)
+    }
+}
+console.log(tempCheck(212))
 
 
-// --------------------2) Create a function that multiplies each number in the array by 5 using a for loop.
-// Use the test variable provided below. Expected outcome: [15, 35, 0, 30, -45] [10, 25, -40, 100, 160]
+
+// // --------------------2) Create a function that multiplies each number in the array by 5 using a for loop.
+// // Use the test variable provided below. Expected outcome: [15, 35, 0, 30, -45] [10, 25, -40, 100, 160]
 
 var myForNumbers1 = [3, 7, 0, 6, -9]
 var myForNumbers2 = [2, 5, -8, 20, 32]
 
+const multiply = (array) => {
+
+    let newArr = []
+    // initializing an empty array
 
+    for(i = 0; i < array.length; i++){
+        // start of for loop
+        newArr.push(array[i] * 5)
+        // adding every element to the empty newArr array multiplied by 5
+    }
+    return newArr
+    // return the new array
+} 
+console.log(multiply(myForNumbers1))
+console.log(multiply(myForNumbers2))
 
 
 
-// --------------------3) Create a function that multiplies each number in the array by 5 using map.
-// Use the test variable provided below. Expected outcome: [435, 30, 140, 80, 10] [40, -35, 0, 30, 60]
+// // --------------------3) Create a function that multiplies each number in the array by 5 using map.
+// // Use the test variable provided below. Expected outcome: [435, 30, 140, 80, 10] [40, -35, 0, 30, 60]
 
 var myMapNumbers1 = [87, 6, 28, 16, 2]
 var myMapNumbers2 = [8, -7, 0, 6, 12]
 
+const mapMult = (array) => {
+    // start of a higher order function taking in an array for its argument
+    return array.map(value => {
+        // using .map to loop through every element 
+        return value * 5
+        // return every element multiplied by 5 
+    })
+}
 
+console.log(mapMult(myMapNumbers1))
+console.log(mapMult(myMapNumbers2))
 
 
 
-// --------------------4) Create a function that removes all the vowels from a string.
-// Use the test variables provided below. Expected output: "HyThrLrnStdnt" "LvJvScrpt"
+// // --------------------4) Create a function that removes all the vowels from a string.
+// // Use the test variables provided below. Expected output: "HyThrLrnStdnt" "LvJvScrpt"
 
 var stringWithVowels1 = "HeyThereLearnStudent"
 var stringWithVowels2 = "ILoveJavaScript"
 
+const onlyVowel = (string) => {
+
+    var spStr = string.split("")
+    // split up all of the strings in the array
+    var result = []
+    // initialize an empty array for later use 
+    for(i = 0; i < spStr.length; i++){
 
+        if(spStr[i] === "a" || spStr[i] === "e" || spStr[i] === "i" || spStr[i] === "o" || spStr[i] === "u"){ 
```

Comment:
Logic is good! Consider a data structure that could allow you store all these values so you can reduce the explicit checking

Reply:
what if there are capitalized vowels?